SEMANTICS.md

```diff
@@ -1100,21 +1100,69 @@ Level 6 supports:
 
 ## Level 7
 
-### Commands
+### Commands and types
 
 Level 7 supports:
-* `print`
-* `ask`
-* `forward`
-* `turn`
-* `is`
-* `sleep`
-* `at random`
-* `add to`
-* `remove from`
-* `if`
-* `if else`
-* `repeat`
+
+<table>
+<thead>
+  <tr>
+    <th>Command</th>
+    <th>Types</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>print</td>
+    <td>integer | string</td>
+  </tr>
+  <tr>
+    <td>ask</td>
+    <td>integer | string</td>
+  </tr>
+  <tr>
+    <td>forward</td>
+    <td>integer</td>
+  </tr>
+  <tr>
+    <td>turn</td>
+    <td>integer</td>
+  </tr>
+  <tr>
+    <td>is</td>
+    <td>string (before `is`) + any (after `is`)</td>
+  </tr>
+  <tr>
+    <td>sleep</td>
+    <td>empty | integer</td>
+  </tr>
+  <tr>
+    <td>at random</td>
+    <td>list</td>
+  </tr>
+  <tr>
+    <td>add to</td>
+    <td>list</td>
+  </tr>
+  <tr>
+    <td>remove from</td>
+    <td>list</td>
+  </tr>
+  <tr>
+    <td>if</td>
+    <td>boolean</td>
+  </tr>
+  <tr>
+    <td>if else</td>
+    <td>boolean</td>
+  </tr>
+  <tr>
+    <td>repeat</td>
+    <td>integer + 'times'</td>
```

Comment:
I am not sure what this means, what is `integer + 'times'` as type?

Reply:
first an integer and then the word times, that's also how I put 'left' and 'right' in possible types used with the command turn in level 1 for example.
(repeat n times) with n an integer is how this command is used, right? 
I think repeat and then an integer does not work? we need the word times behind it?
we could also put times in the command column, but it should be somewhere, not sure which place is the most logical.